packages/react-charting/src/components/VerticalBarChart/VerticalBarChart.base.tsx

```diff
@@ -106,7 +106,15 @@ export class VerticalBarChartBase extends React.Component<IVerticalBarChartProps
 
   public componentDidMount(): void {
     const isChartEmpty =
-      this.state.emptyChart ||
+      this._points.length === 0 ||
+      (d3Max(this._points, (point: IVerticalBarChartDataPoint) => point.y)! <= 0 && !this._isHavingLine);
+    if (this.state.emptyChart !== isChartEmpty) {
+      this.setState({ emptyChart: isChartEmpty });
```

Comment:
Could you clarify that setting it as state variable is preferable than checking at render time.

Reply:
Removing the state reduces number of re-renders, so updated it accordingly.